src/util/color/rgbcolor.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+
+#include <QColor>
+#include <QVariant>
+#include <QtDebug>
+#include <QtGlobal>
+
+#include "util/assert.h"
+#include "util/optional.h"
+
+namespace mixxx {
+
+// Type-safe wrapper for 24-bit RGB color codes without an alpha
+// channel. Code values are implicitly validated upon construction.
+//
+// Apart from the assignment operator this type is immutable.
+class RgbColor {
+  public:
+    // A pure 24-bit 0xxRRGGBB color code with 8-bit per channel
+    // without an an alpha channel.
+    // We are using a separate typedef, because QRgb implicitly
+    // includes an alpha channel whereas this type does not!
+    typedef quint32 code_t;
```

Comment:
What does the `t` stand for?

Reply:
Shorthand for "type", common suffix in C/C++ for types and typedefs. The lowercase type names also pickup the spelling of the corresponding C++ types, i.e. `optional`. Lowercase type names for primitive types or PODs are also not uncommon. This keeps the definitions consistent and intuitive.